c_emulator/riscv_sim.cpp

```diff
@@ -221,147 +202,122 @@ static void read_dtb(const char *path)
+    if ((app->get_name().empty()) && (app->get_parent() != nullptr)) {
+      if (app->get_group() != "SUBCOMMANDS") {
+        out << app->get_group() << ':';
       }
-      insn_limit = val;
-      break;
-    }
-    case OPT_ENABLE_EXPERIMENTAL_EXTENSIONS:
-      fprintf(stderr, "enabling unratified extensions.\n");
-      rv_enable_experimental_extensions = true;
-      break;
-#ifdef SAILCOV
-    case OPT_SAILCOV:
-      sailcov_file = strdup(optarg);
-      break;
-#endif
-    case OPT_TRACE_OUTPUT:
-      trace_log_path = optarg;
-      fprintf(stderr, "using %s for trace output.\n", trace_log_path);
-      break;
-    case '?':
-      print_usage(argv[0], EXIT_FAILURE);
-      break;
     }
-  }
 
-  if (!have_config) {
-    sail_config_set_string(DEFAULT_JSON);
-  }
+    CLI::detail::streamOutAsParagraph(out, make_description(app),
+                                      description_paragraph_width_, "");
+    out << make_usage(app, name);
+    // Suppress the unnecessary "POSITIONALS" options help.
+    // out << make_positionals(app);
+    out << make_groups(app, mode);
+    out << make_subcommands(app, mode);
 
-  if (dtb_file)
-    read_dtb(dtb_file);
+    // Render the footer without any further formatting.
+    out << make_footer(app);
 
-  bool no_elf_file_arg
-      = rvfi || do_print_dts || do_print_isa || do_validate_config;
-  if (optind > argc || (optind == argc && !no_elf_file_arg)) {
-    fprintf(stderr, "No elf file provided.\n");
-    print_usage(argv[0], EXIT_SUCCESS);
+    return out.str();
   }
-  if (!no_elf_file_arg)
-    fprintf(stdout, "Running file %s.\n", argv[optind]);
-  return optind;
+};
+
+static void setup_options(CLI::App &app)
+{
+  app.formatter(std::make_shared<HelpFormat>());
+  // The default formatter gives short options 1/3 of `column_width`
+  // and long options 2/3, which is not great for our options.  Use a
+  // smaller left column width than the default {30} to bring the
+  // short and long options closer together.  This however pushes some
+  // of the option descriptions into the next line, since the long
+  // options now flow into the descriptions column.
+  app.get_formatter()->column_width(20);
+
+  app.add_flag("-p,--show-times", do_show_times, "Show execution times");
+  app.add_flag("--version", do_print_version, "Print model version");
+  app.add_flag("--build-info", do_print_build_info, "Print build information");
+  app.add_flag("--print-default-config", do_print_default_config,
+               "Print default configuration");
+  app.add_flag("--validate-config", do_validate_config,
+               "Validate configuration");
+  app.add_flag("--print-device-tree", do_print_dts, "Print device tree");
+  app.add_flag("--print-isa-string", do_print_isa, "Print ISA string");
+  app.add_flag("--enable-experimental-extensions",
+               rv_enable_experimental_extensions,
+               "Enable experimental extensions");
+  app.add_flag("-N, --use-abi-names", config_use_abi_names,
```

Comment:
```suggestion
  app.add_flag("-N,--use-abi-names", config_use_abi_names,
```

(I guess it works, just for consistency with the other ones.)